Pod/Classes/BetterButton/BetterButton.swift

```diff
@@ -0,0 +1,339 @@
+            }
+
+            // Border
+
+            self.borderColor = borderColor
+            self.borderWidth = borderWidth
+
+            if let borderColor = borderColor {
+                if let highlightedBorderColor = highlightedBorderColor {
+                    self.highlightedBorderColor = highlightedBorderColor
+                }
+                else {
+                    self.highlightedBorderColor = adjustMode.adjustedColor(from: borderColor)
+                }
+            }
+        }
+
+    }
+
+    // Properties
+
+    /// The button shape.
+    public let shape: Shape
+
+    /// The button style.
+    public let style: Style
+
+    /// Use this property to set an image rather than using `setImage:forState`.
+    /// The image will be re-rendered and the button configured appropriately to
+    /// satisfy the provided style.
+    public var iconImage: UIImage? {
+        didSet {
+            if let image = iconImage {
+                let styleAttributes = style.attributes
+                let normalImage = image.tintedImage(color: styleAttributes.foregroundColor)
+                let highlightedImage = image.tintedImage(color: styleAttributes.highlightedForegroundColor)
+                self.setImage(normalImage, for: .normal)
+                self.setImage(highlightedImage, for: .highlighted)
+                self.setImage(highlightedImage, for: .selected)
+            }
+        }
+    }
+
+    /// Setting this property to `true` will replace the button content with an
+    /// activity indicator and disable user interaction. Set to `false` to restore initial button state/behavior.
+    public var isLoading: Bool = false {
+        didSet {
+            if isLoading {
+                activityIndicator.startAnimating()
+            }
+            else {
+                activityIndicator.stopAnimating()
+            }
+
+            self.isUserInteractionEnabled = !isLoading
+            self.imageView?.isHidden = isLoading
+            self.titleLabel?.isHidden = isLoading
+        }
+    }
+
+    fileprivate let activityIndicator: UIActivityIndicatorView = {
+        let view = UIActivityIndicatorView(activityIndicatorStyle: .white)
+        view.hidesWhenStopped = true
+        return view
+    }()
+
+    /// Create a new instance.
+    ///
+    /// - Parameters:
+    ///   - shape: The desired button shape.
+    ///   - style: The desired button style.
+    public init(shape: Shape, style: Style) {
+        self.shape = shape
+        self.style = style
+        super.init(frame: .zero)
+
+        addSubview(activityIndicator)
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("BetterButton does not yet support the use of Interface Builder.")
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        activityIndicator.center = CGPoint(x: bounds.midX, y: bounds.midY)
```

Comment:
Some times the loading indicator wants to just replace the icon, or to be aligned to the left of the text and for the text to not be hidden. Might be able to have an `enum ActivityIndicatorStyle`?